app/src/main/java/ayds/lisboa/songinfo/moredetails/fulllogic/OtherInfoWindow.kt

```diff
@@ -0,0 +1,199 @@
+        val artistUrl = artist.getArtistUrl()
+        artistUrl.setOpenUrlButton()
+    }
+
+    private fun getArtistAsJsonObject(): JsonObject {
+        val callResponse = lastFMAPI.getArtistInfo(artistName).execute()
+        return getArtistFromCallResponse(callResponse)
+    }
+
+    private fun getArtistFromCallResponse(callResponse: Response<String>): JsonObject{
+        val gsonObject = Gson()
+        val jObjFromGson = gsonObject.fromJson(callResponse.body(), JsonObject::class.java)
+        return jObjFromGson[ARTIST].asJsonObject
+    }
+
+    private fun JsonObject.getArtistBioContent() = this[BIO].asJsonObject[CONTENT]
+
+    private fun JsonObject.getArtistUrl() = this[URL]
+
+    private fun artistBioAsHTML(artistBioContent: String): String {
+        val artistBioContentReformatted = artistBioContent.reformatArtistBio()
+        return textToHtml(artistBioContentReformatted)
+    }
+
+    private fun String.reformatArtistBio() = this.replace("\\n", "\n")
+
+    private fun JsonElement.setOpenUrlButton() {
+        val urlString = this.asString
+        openUrlButton.setOnClickListener {
+            val intent = Intent(Intent.ACTION_VIEW)
+            intent.data = Uri.parse(urlString)
+            startActivity(intent)
+        }
+    }
+
+    private fun textToHtml(text: String): String {
+        val builder = StringBuilder()
+        val textWithBold = textAsBold(text)
+
+        builder.append(HTML_OPENING_TAG)
+        builder.append(textWithBold)
+        builder.append(HTML_CLOSING_TAG)
+
+        return builder.toString()
+    }
+
+    private fun textAsBold(text: String) : String {
+        return text
+            .replace("'", " ")
+            .replace("\n", "<br>")
+            .replace(
+                "(?i)$artistName".toRegex(),
+                "<b>" + artistName.uppercase(Locale.getDefault()) + "</b>"
+            )
+    }
+
+    companion object {
+        const val ARTIST_NAME_EXTRA = "artistName"
+    }
+
+    private fun updateArtistInfo() {
+        val artist = searchArtist()
+        val bioContent = obtainBioContent(artist)
+        val artistInfoHTML = artistBioAsHTML(bioContent)
+        setTextPane(artistInfoHTML)
+        setURLButton()
+    }
+
+    private fun obtainBioContent(artist: Artist): String {
+        if (artist is Artist.ArtistData){
+            return if (artist.isLocallyStored) {
+                "$PREFIX${artist.artistBioContent}"
+            } else artist.artistBioContent
+        }
+        return "No Results"
+    }
+
+    private fun searchArtist(): Artist {
+        var artist: Artist?
+        artist = dataBase.getArtist(artistName)
+        if (artist is Artist.EmptyArtist) {
+            try {
+                artist = getArtistFromLastFMAPI()
+                saveArtistInfo(artist)
+            } catch (e: Exception) {
+                artist = null
+            }
+        }
+        else {
+            markArtistAsLocal(artist as Artist.ArtistData)
+        }
```

Comment:
Se puede evitar el cast usando when
```suggestion
        when (artist) {
            Artist.EmptyArtist -> try {
                    artist = getArtistFromLastFMAPI()
                    saveArtistInfo(artist)
                } catch (e: Exception) {
                    artist = null
                }
            is  Artist.ArtistData -> markArtistAsLocal(artist as)
        }
        
```